nf_core/download.py

```diff
@@ -1148,6 +1232,15 @@ def singularity_image_filenames(self, container):
         # Add file extension
         out_name = out_name + extension
 
+        # Trim potential registries from the name for consistency.
+        # This will allow pipelines to work offline without symlinked images,
+        # if docker.registry / singularity.registry are set to empty strings at runtime, which can be included in the HPC config profiles easily.
+        if self.registry_set:
+            # Create a regex pattern from the set of registries
+            trim_pattern = "|".join(f"{re.escape(registry)}-?" for registry in self.registry_set)
+            # Use the pattern to trim the string
+            out_name = re.sub(f"^{trim_pattern}", "", out_name)
```

Comment:
I will check. 

Likely not, because hardly any function documentation is still up-to-date there 🫠. The Seqera people have quite ambitious goals for refactoring this functionality (_mainly [#2408](https://github.com/nf-core/tools/issues/2408), but also [#2406](https://github.com/nf-core/tools/issues/2406), [#2407](https://github.com/nf-core/tools/issues/2407)_) and therefore I was never sure how much effort I should still put into the current code (e.g. also typing the functions).